### ISSUE TITLE
Pod Provider Admin Actions

### ISSUE THREAD

**Issue by kjetilk** [RubenVerborgh/solid-server-architecture#29]
We also need a bunch of actions for Pod Providers that is also outside of LDP, like pod provisioning, quota administration, contact users, delete Pods that doesn't conform to the ToS.

**Comment by RubenVerborgh** [comment]
Idem as #28; not sure if sufficiently complex that these warrant a diagram.

**Comment by kjetilk** [comment]
Getting nearer, I just want to see the quota hooked up in the main diagram, where do you intercept write operations?

**Comment by RubenVerborgh** [comment]
> I just want to see the quota hooked up in the main diagram

You actually would not see the connection in the WAC/LDP component diagram, because those components only depend on interfaces. They would not know that they are hooked up to a quota-aware store or not (which is a good thing, of course).

A deployment diagram might show it.

How it works is that an `Operation` is constructed using a `ResourceStore` (by the `OperationFactory`, which in turn gets it as a constructor argument). So it is up to the general component assembler to inject the right dependency.

In the quota case, `OperationFactory` would receive a certain `ResourceStore` instance. If that `ResourceStore` instance also happens to implements `SizeReporter`, then this is the component passed to `AdministrationApi`. If not, a `SizeReporter` (possibly `SizeCache`) needs to be created around the _specific_ `ResourceStore`, and then passed to `AdministrationApi`.

So long story short: `OperationFactory` and `AdministrationApi` are passed the same `ResourceStore` (or a wrapper around it).

**Comment by kjetilk** [comment]
OK, allright. I think this would have been a lot clearer with a traits based architecture... :-)

**Comment by RubenVerborgh** [comment]
Resolving this for now as the scope is LDP+ACL.
